os/fs/smartfs/smartfs.h

```diff
@@ -311,6 +316,7 @@ struct smartfs_ofile_s {
 #ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
 	uint8_t *buffer;			/* Sector buffer to reduce writes */
 	uint8_t bflags;				/* Buffer flags */
+	mode_t mode;
```

Comment:
good, I forgot regarding mode, but this also part of entry.

Reply:
Can take care of this before merging.
Some more critical logical changes first.